### PR TITLE
Add params to CTA redirect URL

### PR DESCRIPTION
**Pull request by mcbattirola**
This PR updates the redirect URL when a user clicks a CTA and adds additional params. It also updates the button to match the way we do link buttons on other components (`as="a"` instead of window.open on click).

The change is kind of big due to snapshots and having to add context around the stories that used the button, but the only relevant changes are in `web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx`.

See details in this [slab doc](https://gravitational.slab.com/posts/utm-parameters-1rlklvez#h9tgh-in-product-ut-ms).

### CODE REVIEW

File: web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx [hunk 1]
```diff
@@ -29,17 +30,24 @@ export type Props = {
   [index: string]: any;
 };
 
-const salesUrl = 'https://goteleport.com/signup/enterprise/';
+const salesUrl = 'https://goteleport.com/r/upgrade-team';
```
Comment (kimlisa) [inline web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx:33]: think we prefer screaming snake casing for js consts

```suggestion
const SALES_URL = 'https://goteleport.com/r/upgrade-team';
```

File: web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx [hunk 2]
```diff
@@ -29,17 +30,24 @@ export type Props = {
...
+  const version = ctx.storeUser.state.cluster.authVersion;
+  const isEnterprise = ctx.isEnterprise;
+
   function handleClick() {
     userEventService.captureCtaEvent(event);
-    window.open(salesUrl, 'blank');
+    window.open(
+      `${salesUrl}?${getParams(version, isEnterprise, event)}`,
+      'blank'
+    );
```
Comment (kimlisa) [inline web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx:50]: afaik, we don't use `window.open` for a button click out, we instead do something like this: https://github.com/gravitational/teleport/blob/42c95bfd6b70484936fccd8432df5f570c41ef78/web/packages/teleport/src/Discover/Shared/ActionButtons.tsx#L47

Comment (mcbattirola) [reply web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx]: Thanks, updated in https://github.com/gravitational/teleport/pull/26086/commits/c67b9bbb4121df7e378271ca2728ade32d4a2469.

**Review by ryanclark — APPROVED** [review]

File: web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx [hunk 3]
```diff
@@ -29,36 +30,49 @@ export type Props = {
...
     userEventService.captureCtaEvent(event);
-    window.open(salesUrl, 'blank');
   }
 
   return (
-    <StyledButton onClick={handleClick} {...rest}>
+    <ButtonBorder
+      as="a"
+      target="blank"
+      kind="primary"
```
Comment (kimlisa) [inline web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx:53]: hrm we imported `ButtonBorder`, but then use kind `primary`, would just importing `ButtonPrimary` and using that work?

**Review by kimlisa — APPROVED** [review]

File: web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx (hunk 3 above)
Comment (mcbattirola) [reply web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.tsx:53]: We can, actually. Updated, thanks!

**Pull request merged**